src/pybind/mgr/dashboard/module.py

```diff
@@ -751,6 +752,133 @@ def get_perf_schema(self, **args):
+                for osd in osd_map['osds']:
+                    osd_id = str(osd['osd'])
+                    counter_data = global_instance().get_counter('osd', osd_id, path)
+                    osd_id = "osd-" + osd_id
+                    result[osd_id] = counter_data[path]
+                return result
+
+            # Method to get the overall statistical distribution of a given perf counter.
+            @cherrypy.expose
+            @cherrypy.tools.json_out()
+            def get_counter_stats(self, path):
+                
+                data = self.get_counter_allosd(path)
+                counter_stats_data = {
+                    "perf_counter" : path,
+                    "osds" : data,
+                    "min" : [],
+                    "max" : [],
+                    "avg" : [],
+                    "stdev" : [],
+                    "stdevs" : {
+                        "plus_onehalf" : [],
+                        "plus_half" : [],
+                        "minus_half" : [],
+                        "minus_onehalf" : []
+                    },
+                    "osds_out_sync" : []
+                    }
+
+                result = []
+                sync_data = []
+                osd_count = len(data)
+
+                ref_osd = None
+                epoch_set = {}
+                for osd_id in data:
+                    epoch = str(data[osd_id][0][0])
+                    if epoch not in epoch_set:
+                        epoch_set[epoch] = []
+                    epoch_set[epoch].append(osd_id)
+                max_sync = 0
+                for epoch in epoch_set:
+                    sync_count = len(epoch_set[epoch])
+                    if sync_count > max_sync:
+                        ref_osd = epoch_set[epoch][0]
+                        max_sync = sync_count
+
+                for i in range(0,len(data[ref_osd])):
+
+                      epoch = data[ref_osd][i][0]
+                      values = []
+                      outsync_osds = []
+
+                      for osd_id in data:
+                            if data[osd_id][i][0] != epoch:
+                                outsync_osds.append(osd_id)
+                            values.append(data[osd_id][i][1])
+
+                      result.append(
+                            [epoch, values])
+                      counter_stats_data["osds_out_sync"].append(
+                            [epoch, outsync_osds])
+
+                for tseq in result:
+                    
+                    length = len(tseq[1])
+                    avg = sum(tseq[1])/length
+                    counter_stats_data["avg"].append([tseq[0], avg])
+                    diff = [(x-avg)**2 for x in tseq[1]]
+                    var = sum(diff)/length
+                    stdev = math.sqrt(var)
+                    counter_stats_data["stdev"].append([tseq[0], stdev])
+                    counter_stats_data["stdevs"]["plus_onehalf"].append([tseq[0], avg+(1.5*stdev)])
```

Comment:
what are these plots for: plus_onehalf, plus_half, ...?

Reply:
These plots are to refer to the standard deviations around average, plus_onehalf = Average + (1.5 *stdev), plus_half = Average + (0.5*stdev), minus_half  = Average - (0.5*stdev) and minus_onehalf = average - (1.5*stdev). This way distribution of latency values on both sides of the average value can be seen for the performance of OSDs.